crates/router/src/connector/plaid/transformers.rs

```diff
@@ -0,0 +1,482 @@
+}
+
+impl TryFrom<&PlaidRouterData<&types::PaymentsAuthorizeRouterData>> for PlaidPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: &PlaidRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
+        match item.router_data.request.payment_method_data.clone() {
+            domain::PaymentMethodData::OpenBanking(ref data) => match data {
+                domain::OpenBankingData::OpenBankingPIS { .. } => {
+                    let amount = item.amount;
+                    let currency = item.router_data.request.currency;
+                    // This reference is supposed to be under 18 chars in length, so unable to use payment_id
+                    let reference = "SomeRef".to_string();
+                    let recipient_val = item
+                        .router_data
+                        .connector_meta_data
+                        .as_ref()
+                        .ok_or(errors::ConnectorError::MissingRequiredField {
+                            field_name: "connector_customer",
+                        })?
+                        .peek()
+                        .clone();
+
+                    let recipient_type =
+                        serde_json::from_value::<types::MerchantRecipientData>(recipient_val)
+                            .change_context(errors::ConnectorError::ParsingFailed)?;
+                    let recipient_id = match recipient_type {
+                        types::MerchantRecipientData::ConnectorRecipientId(id) => {
+                            Ok(id.peek().to_string())
+                        }
+                        _ => Err(errors::ConnectorError::NotSupported {
+                            message: "recipient_id not found for Plaid".to_string(),
+                            connector: "plaid",
+                        }),
+                    }?;
+
+                    Ok(Self {
+                        amount: PlaidAmount {
+                            currency,
+                            value: amount,
+                        },
+                        reference,
+                        recipient_id,
+                        schedule: None,
+                        options: None,
+                    })
+                }
+            },
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
+
+impl TryFrom<&types::PaymentsSyncRouterData> for PlaidSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsSyncRouterData) -> Result<Self, Self::Error> {
+        match item.request.connector_transaction_id {
+            types::ResponseId::ConnectorTransactionId(ref id) => Ok(Self {
+                payment_id: id.clone(),
+            }),
+            _ => Err(
+                errors::ConnectorError::NotImplemented("ResponseId for Plaid".to_string()).into(),
```

Comment:
Replace the `ConnectorError::NotSupported` with `ConnectorError::MissingConnectorTransactionID`